### PR TITLE
MYFACES-4588

### PR DESCRIPTION
**Pull request by werpu**
Fixes the issue that getWindowId did not look into the document in the correct order.
Also added tests for the case and added tests for the file upload given from the Tobago guys (which I forgot to add to git)

### CODE REVIEW

**Comment by werpu** [comment]
Seems like we have an integration test build issue, apparently the chrome driver has a version mismatch with chrome version in the build system. Wont merge until everything passes. But the fix is fine (see the test protocols) and independent of the failing integration tests.

**Comment by bohmber** [comment]
Unfortunately there is not better fix at the moment

**Comment by bohmber** [comment]
The version should be 110.0.5481.77

File: api/src/client/typescript/faces/impl/AjaxImpl.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ import {
     EMPTY_STR,
     CTX_PARAM_MF_INTERNAL,
     NAMED_VIEWROOT,
-    NAMING_CONTAINER_ID
+    NAMING_CONTAINER_ID, $nsp
```
Comment (volosied) [inline api/src/client/typescript/faces/impl/AjaxImpl.ts:57]: Should it be on a new line?

**Comment by volosied** [comment]
+1  

Tested locally and the fix works! Thanks :)

File: api/src/client/typescript/faces/impl/AjaxImpl.ts (hunk 1 above)
Comment (werpu) [reply api/src/client/typescript/faces/impl/AjaxImpl.ts]: fixed

**Comment by werpu** [comment]
ci build still failing, but given this is unrelated to my fix and also on the master branch, I guess we can merge now!

**Comment by bohmber** [comment]
Build is running successful now with the correct chromeDriverVersion

**Comment by werpu** [comment]
Thanks for fixing it!

**Pull request merged**